File: golang/0643.go
func findMaxAverage(nums []int, k int) float64 {
	length := len(nums)
	sum := 0
	for i := 0; i < k; i++ {
		sum += nums[i]
	}

	maxx := sum
	for i := k; i < length; i++ {
		sum = sum - nums[i-k] + nums[i]
		if sum > maxx {
			maxx = sum
		}
	}
	return float64(maxx) / float64(k)
}

File: rust/0643.rs
impl Solution {
    pub fn find_max_average(nums: Vec<i32>, k: i32) -> f64 {
        let mut sum = 0;
        for i in 0..k {
            sum += nums[i as usize];
        }
        let mut res = sum;
        let length = nums.len();
        let k = k as usize;
        for i in (k as usize)..length {
            sum += nums[i] - nums[i-k];
            if sum > res {
                res = sum
            }
        }

        res as f64 / k as f64
    }
}
